### ISSUE TITLE
No-code frame to drop airdrop NFT to eligible user

### ISSUE THREAD

**Issue by Ramitphi** [unlock-protocol/unlock#13366]
### Description

Our idea is to buil a no-code frame generator tooling which can allow any non-tech person to create the frames on the go for functionality of airdropping.
We are adding the layer of engagement/participation to it, which means one people who attended the certain meeting can claim the NFT which we airdrop them using Unlock Porotcol

### Website URL

https://huddle01.com/

### Twitter

@phi_ramit

### Github

https://github.com/Huddle01

### LinkedIn

https://www.linkedin.com/in/ramit-agarwal-46547316b/

### Discord handle

ramitphi

### Completion time

3 weeks

### Team

Team Details

Ayush Ranjan (CEO) :  https://www.linkedin.com/in/ranjan18/
Susmit Lavania(CTO) : https://www.linkedin.com/in/susmit-lavania-4b9558107/
Daria Saji (Product Manager): https://www.linkedin.com/in/dariasaji/
Ramit Agarwal( Tech PM):https://www.linkedin.com/in/ramit-agarwal-46547316b/

### Milestones

Milestone 1 (200 UDT)
- User able to login
- Able to upload image of their choice which will be minted as NFT
- Able to deploy the lock

Milestone 2 (300UDT)
- Frame creator button
- People can create the frame in one click
- They get an unique link which they can embed it in their casts

### Requested grant amount

Other amount, or not requesting UDT

### Anything else you think we should know, including why you've requested a particular grant amount

We are requesting for 500 UDT as we believe working with frames would require certain expertise and added dev hours with their learning curve. On the other side, we believe it will be viral product as it solves out the problem of most of the community manager to handle NFT drops.
We will be also sponsoring gas fees for claim process,  to provide the gas-less experience to the end user

### Is your work Open Source

No, my project does not use an Open Source license

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ccarfi** [comment]
Love this idea! @Ramitphi, where would the image be stored?

`Able to upload image of their choice which will be minted as NFT
`

**Comment by Calla-Ji** [comment]
Hi @Ramitphi 

Thank you for putting this up!
Will users/ creators be able to own the contract through which this is dropped?
So, for example, I made my NFT on Unlock, and now I want to use your application to make it claimable on Warpcaster. Is that a thing?

Thank you in advance <3

**Comment by Ramitphi** [comment]
> Love this idea! @Ramitphi, where would the image be stored? Is this using the typical Unlock-powered storage (the same as when someone creates a membership today from the Unlock Dashboard)? or something else?
> 
> `Able to upload image of their choice which will be minted as NFT `

I was thinking to store ipfs the way it usually happens might use pinata or so, but if any suggestion on the storage part would be happy to look into them

**Comment by Ramitphi** [comment]
> Hi @Ramitphi
> 
> Thank you for putting this up! Will users/ creators be able to own the contract through which this is dropped? So, for example, I made my NFT on Unlock, and now I want to use your application to make it claimable on Warpcaster. Is that a thing?
> 
> Thank you in advance <3

Yes they will own the contract, we will allow the creation of lock from our webpage itself.

- User deploy the lock from our frontend
- They have the deployed lock 
- They can now select the Huddle01 meeting room for which they want to airdrop NFT 
- A frame link will be created which can be used in warpcast directly

No hassle of collecting wallet, creating drops & gas less experience at once place, on the other hand user who wana claim just have to do single click all the checks happen in backend

**Comment by Calla-Ji** [comment]
> > Love this idea! @Ramitphi, where would the image be stored? Is this using the typical Unlock-powered storage (the same as when someone creates a membership today from the Unlock Dashboard)? or something else?
> > `Able to upload image of their choice which will be minted as NFT `
> 
> I was thinking to store ipfs the way it usually happens might use pinata or so, but if any suggestion on the storage part would be happy to look into them

I have a personal preference for Arweave storage @Ramitphi if that is possible. I feel it relates a bit better to the idea of decentralized storage <3

**Comment by Calla-Ji** [comment]
> > Hi @Ramitphi
> > Thank you for putting this up! Will users/ creators be able to own the contract through which this is dropped? So, for example, I made my NFT on Unlock, and now I want to use your application to make it claimable on Warpcaster. Is that a thing?
> > Thank you in advance <3
> 
> Yes they will own the contract, we will allow the creation of lock from our webpage itself.
> 
> * User deploy the lock from our frontend
> * They have the deployed lock
> * They can now select the Huddle01 meeting room for which they want to airdrop NFT
> * A frame link will be created which can be used in warpcast directly
> 
> No hassle of collecting wallet, creating drops & gas less experience at once place, on the other hand user who wana claim just have to do single click all the checks happen in backend

That's incredible and thank you for breaking it down so well! 
You got my support for this one <3

**Comment by Ramitphi** [comment]
@ccarfi just curious to know what are the next steps, once we have opened the issue

**Comment by Alanatheweb3girl** [comment]
@Ramitphi how about bringing it up in the DAO Call next week (cause this week is canceled) and getting a temperature check on it during the grants review section of the call? After that open your Snapshot proposal. Get it hopefully through and start building...

**Comment by Ramitphi** [comment]
works with me, will bring it on table for discussion in the next DAO Call

**Comment by ccarfi** [comment]
`@ccarfi just curious to know what are the next steps, once we have opened the issue`

@ramitphi The grant process is run by the DAO, so it's really up to you all! The process in the past is outlined here

[https://docs.unlock-protocol.com/governance/grants-bounties/udt-grantee-handbook/](https://docs.unlock-protocol.com/governance/grants-bounties/udt-grantee-handbook/
)

Again, however, that's a recommended process. If you think there's been sufficient conversation, and you feel there is sufficient support, you could go straight to an onchain vote (if you choose to propose it), for example. 

Your work is very well known to the community, as is your ability to deliver (and the fact that this has been prototyped already) de-risks it substantially from my opinion.

**Comment by Ramitphi** [comment]
thanks for the revert @ccarfi.

created the proposal on snapshot for the same, once it's done will move it to on-chain voting
https://snapshot.org/#/unlock-protocol.eth/proposal/0x85deb825eeb454442c505887d9c19454d4e9fd8a5005eee720b567e173ea7a98

**Comment by CaptainSouthpaw** [comment]
with regards to making this an open source solution, I will delegate my voting power to someone who would vote yes if the proposal is changed thusly.

**Comment by Calla-Ji** [comment]
Hi @Ramitphi :)

Since your proposal got defeated, I re-initiated a discussion about it today during the DAO Call as I had seen Blahkheart's comment on Snapshot, stating the following:

`I love the project idea, but requesting for 500UDT for a project that is not Open Source for community benefit from learnings is a no for me`

The same comments were made today during the call. Would you be willing to adjust your proposal toward an open-source contribution?

Best
Stella*

**Comment by ccarfi** [comment]
Re-opened as an OPEN-SOURCE proposal.
https://snapshot.org/#/unlock-protocol.eth/proposal/0xd6b416b5b63fdcdfd915e28afc1735f4e39a5ac3eb14a85074b5ea8b5fe49e04
